tests/view/treewalker.js

```diff
@@ -0,0 +1,628 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* bender-tags: view */
+
+'use strict';
+
+import Document from '/ckeditor5/engine/view/document.js';
+import Element from '/ckeditor5/engine/view/element.js';
+import ContainerElement from '/ckeditor5/engine/view/containerelement.js';
+import Text from '/ckeditor5/engine/view/text.js';
+import TreeWalker from '/ckeditor5/engine/view/treewalker.js';
+import Position from '/ckeditor5/engine/view/position.js';
+import Range from '/ckeditor5/engine/view/range.js';
+import CKEditorError from '/ckeditor5/utils/ckeditorerror.js';
+
+describe( 'TreeWalker', () => {
+	let doc, root, img1, paragraph, bold, textBa, charR, img2, charX;
+	let rootBeginning, rootEnding;
+
+	before( () => {
+		doc = new Document();
+		root = doc.createRoot( document.createElement( 'div' ) );
+
+		// root
+		//  |- img1
+		//  |- p
+		//     |- b
+		//     |  |- B
+		//     |  |- A
+		//     |
+		//     |- R
+		//     |
+		//     |- img2
+		//     |
+		//     |- X
+
+		textBa = new Text( 'ba' );
+		bold = new Element( 'b', [], [ textBa ] );
```

Comment:
It was, then I thought this element doesn't really have attributes so I can change it to Element https://github.com/ckeditor/ckeditor5-engine/pull/492/commits/3163e4f4b6fb5fd666997007b5d6a68684413e64 :)